packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -76,6 +76,8 @@ const TSErrors = Object.freeze({
   IndexSignatureHasStatic: "Index signatures cannot have the 'static' modifier",
   OptionalTypeBeforeRequired:
     "A required element cannot follow an optional element.",
+  LabeledElementBeforeUnlabeled:
+    "An unlabeled element cannot follow a labeled element.",
```

Comment:
TypeScript throws `Tuple members must all have names or all not have names`. However, it accepts this code:
```
type B = [A, x: C];
```

[Playground Link](https://www.typescriptlang.org/play?ts=4.0.0-dev.20200627&ssl=2&ssc=20&pln=2&pc=1#code/C4TwDgpgBAglC8UDaBnYAnAlgOwOYF0BuAWAChRIoAhBZGAGigA8AuaooA)

@weswigham / @DanielRosenwasser Which is the correct behavior?

Reply:
Uhh.. we _shouldn't_ accept that (but it is just a grammar error, not a parse error). Care to open an issue?